L1Trigger/Phase2L1ParticleFlow/interface/SimpleCalibrations.h

```diff
@@ -0,0 +1,174 @@
+          std::vector<double> ptMax = cpset.getParameter<std::vector<double>>("ptMax");
+          ptMaxs.insert(ptMaxs.end(), ptMax.begin(), ptMax.end());
+        } else {
+          ptMaxs = std::vector<float>(etaBins.size(), 1e6);
+        }
+      }
+    }
+
+    bool empty() const { return etas.empty(); }
+    float operator()(float pt, float abseta) const {
+      for (unsigned int i = 0, n = etas.size(); i < n; ++i) {
+        if (pt > ptMaxs[i])
+          continue;
+        if (abseta < etas[i]) {
+          if (pt < ptMins[i])
+            return (pt / ptMins[i]) * (ptMins[i] - offsets[i]) / scales[i];
+          return (pt - offsets[i]) / scales[i];
+        }
+      }
+      return pt;
+    }
+
+  protected:
+    std::vector<float> etas, scales, offsets, ptMins, ptMaxs;
+  };
+  class SimpleCorrHad {
+  public:
+    SimpleCorrHad() {}
+    SimpleCorrHad(const edm::ParameterSet &iConfig, const std::string &name) {
+      if (iConfig.existsAs<edm::ParameterSet>(name)) {
+        edm::ParameterSet cpset = iConfig.getParameter<edm::ParameterSet>(name);
+        std::vector<double> etaBins = cpset.getParameter<std::vector<double>>("etaBins");
+        std::vector<double> emfBins = cpset.getParameter<std::vector<double>>("emfBins");
+        std::vector<double> offset = cpset.getParameter<std::vector<double>>("offset");
+        std::vector<double> scale = cpset.getParameter<std::vector<double>>("scale");
+        etas.insert(etas.end(), etaBins.begin(), etaBins.end());
+        emfs.insert(emfs.end(), emfBins.begin(), emfBins.end());
+        scales.insert(scales.end(), scale.begin(), scale.end());
+        offsets.insert(offsets.end(), offset.begin(), offset.end());
+        if (cpset.existsAs<std::vector<double>>("ptMin")) {
+          std::vector<double> ptMin = cpset.getParameter<std::vector<double>>("ptMin");
+          ptMins.insert(ptMins.end(), ptMin.begin(), ptMin.end());
+        } else {
+          float ptMin = cpset.existsAs<double>("ptMin") ? cpset.getParameter<double>("ptMin") : 0;
+          ptMins = std::vector<float>(etaBins.size(), ptMin);
+        }
+        if (cpset.existsAs<std::vector<double>>("ptMax")) {
+          std::vector<double> ptMax = cpset.getParameter<std::vector<double>>("ptMax");
+          ptMaxs.insert(ptMaxs.end(), ptMax.begin(), ptMax.end());
+        } else {
+          ptMaxs = std::vector<float>(etaBins.size(), 1e6);
+        }
+      }
+    }
+
+    bool empty() const { return etas.empty(); }
+    float operator()(float pt, float abseta, float emf) const {
+      unsigned int i = 0, n = emfs.size();
+      // step 0 get to the right pt bin
+      while (i < n && pt > ptMaxs[i])
+        i++;
+      // step 1: find the first emf bin for this eta
+      while (i < n && abseta > etas[i])
+        i++;
+      if (i == n) {
+        //printf("for pt %7.2f eta %4.2f emf %4.2f will not apply any correction (no eta bin found)\n", pt, abseta, emf);
```

Comment:
it is there because at times during development and debugging  it is useful to uncomment it.
if you prefer it to be hidden in any other way like `if(0) { ... }` or a `#if 0` I could do it, but in practice they're both less readable

Reply:
If commented-out code should be kept, the reason must be indicated in another comment.

CMSSW uses MessageLogger for messaging. These messages could be converted to LogDebug, which are skipped during normal compilation.